### PR TITLE
Add word count generator

### PR DESCRIPTION
**Pull request by pvcarrera**
In the list of exercises that need a generator #396 , I saw word-count. Once I started working on it, I found the template `exercises/word-count/example.tt ` but I couldn't find any generator associated to it, so I decided to modify the current template and create the generator that you can see in this PR.
 
The `canonical-data.json` file from x-common has less examples that the current version of `word_count_test.rb` Should I add these examples to x-common?

### CODE REVIEW

**Comment by kotp** [comment]
If the prior non-generated version of the test had more tests than what is in x-common, then we need to review the discussions that lead to that difference.

Do you feel like the additional tests are something that is Ruby specific?  Do you feel that the additional tests may be redundant or a good addition to other languages?

**Comment by kotp** [comment]
BTW: Thanks for jumping in here.

**Comment by pvcarrera** [comment]
The additional cases are: 
```
test_count_everything_just_once
test_handles_cramped_lists
test_with_apostrophes
test_with_quotations
```
I think this are valid test that cover cases that are not covered in the x-common cases

**Comment by Insti** [comment]
@pvcarrera create a PR in [exercism/x-common](https://github.com/exercism/x-common) adding the additional cases to the canonical-data.json file.

File: lib/word_count_cases.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  def name
+    'test_%s' % description.tr(' ', '_')
+  end
+
+  def object_under_test
+    "Phrase.new('#{input}')"
+  end
+
+  def skipped?
+    index > 0
```
Comment (kotp) [inline lib/word_count_cases.rb]: Or even `index.notzero?`   No action needed, just an annotation.

**Comment by pvcarrera** [comment]
I think I need to update this branch/PR: Merge #451, then remove the generate-word-count scripts. Also, the generated tests don't follow the project style rules. I ran `rubocop -a` after the test cases where generated but, after the changes I did in #454, I realised that the test must be generated with the correct format.

**Comment by kotp** [comment]
Don't worry about updating the branch against 451, you should be able to rebase against master.

There is another change (unsure if it is in yet, though) that sets style enforcement/checks back to 0.  But this would also be on master.

The process for doing this would be on your fork of the project...

(Not tested, just written out here as a guide)

```bash
git checkout master
git pull origin master
git checkout add-word-count-generator
git rebase master
```

That should move your branch to the latest master branch.

**Comment by pvcarrera** [comment]
Done, I rebased, squashed and forced push. 

I'm not sure if this is open to discussion, but this is my opinion about PRs workflow:

Rebasing an open PR rewrites the PR history, this means that comments are not synchronized with changes anymore. I don't usually rebase/force push once a PR is open, on the other hand, I add commits and merge master in the branch as many times as necessary. Once the PR is ready, I use the GitHub Squash and merge option. This reduces the PR to one commit but still keeps a link to the closed PR request where you can find the original code with its comments.

**Comment by kotp** [comment]
I was only stating about rebasing from master, since I changed how the Rubocop is behaving.

Thanks for the notes on the PR workflow.  It is very similar to how things are normally done here, though we reduce to "a few" if appropriate rather than always a single commit.  

I did not realize we were ready for that, but you doing so, definitely signals that we are ready for "final approach" on this.

**Comment by kotp** [comment]
#451 Common generation script is now in master.

**Comment by pvcarrera** [comment]
Cool, this PR is already up to date with #451

**Comment by kotp** [comment]
OK, I attempted to do a generate, might me doing something wrong since it changed recently.
Is that working for you?

**Comment by kotp** [comment]
I am using `ASSIGNMENT=word-count make test-assignment` to test it locally, it is coughing due to the line 68 in the tests.

```text
running tests for: word-count
/tmp/word-count.CuENNcRPR7/word_count_test.rb:68: syntax error, unexpected tIDENTIFIER, expecting ')'
    phrase = Phrase.new('First: don't laugh. Then: don't cry.')
```

**Comment by bmulvihill** [comment]
Looks like its due a to a use of single quotes as you mentioned in https://github.com/exercism/xruby/pull/463.  I pulled locally and changed the object_under test method as so 
```ruby
  def object_under_test
    "Phrase.new(\"#{input}\")"
  end
```

Output:
screen shot 2016-10-18 at 3 35 02 pm

**Comment by Insti** [comment]
> `"Phrase.new(\"#{input}\")"`

This will still break if the string contains double quotes. 
Much better to find a solution that correctly escapes quote characters.

**Comment by pvcarrera** [comment]
@Insti The input string comes from a json file, example: `"input": "Joe can't tell between 'large' and large."` I guess if one of the strings contains a double quote it'll come scaped, something like: `"input": "Joe can't tell between \"large\" and large."`

**Comment by Insti** [comment]
It's still a problem because the json is parsed before we try to insert it.

Example:
```ruby
require 'json'
json = { :example => 'Contains " double quote' }.to_json
puts json
input = JSON.parse(json)['example']
puts %Q(Quote not escaped "#{input}")
```
Output:
```
{"example":"Contains \" double quote"}
Quote not escaped "Contains " double quote"
```

This might work though:
```
%Q(Quote now escaped #{input.to_json})
```
(Note that the surrounding quotes are now provided by the .to_json method.)

File: lib/word_count_cases.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ def name
   end
 
   def object_under_test
-    %Q(Phrase.new("#{input}"))
+    %Q(Phrase.new(#{input.to_json}))
```
Comment (Insti) [inline lib/word_count_cases.rb]: Sorry to be nitpicky, but..
I think I prefer using `inspect` here, since it's not REALLY json that we want.

**Comment by Insti** [comment]
I think this is pretty much ready now. :+1:
@kotp?

File: lib/word_count_cases.rb (hunk 2 above)
Comment (pvcarrera) [inline lib/word_count_cases.rb]: Yep, I agree `inspect` reveals the intention better than `to_json`. Thanks

**Comment by Insti** [comment]
Thanks for all your work on this @pvcarrera :heart:

**Comment by pvcarrera** [comment]
I'm really happy to help, thanks a lot for all of your feedback 😍

**Comment by kotp** [comment]
Thanks @pvcarrera :tophat: to @Insti

**Pull request merged**